tdrs-backend/tdpservice/users/api/login.py

```diff
@@ -62,19 +62,51 @@ def handle_user(self, request, id_token, decoded_payload):
         if "token" not in request.session:
             request.session["token"] = id_token
 
+        # Authenticate users with the unique "subject" `sub` UUID from the payload.
+        sub = decoded_payload["sub"]
+        email = decoded_payload["email"]
+
+        # First account for the initial superuser
+        if (su_username := os.environ.get('DJANGO_SU_NAME')) and su_username == email:
+            # If this is the initial login for the initial superuser,
+            # we must authenticate with their username since we have yet to save the
+            # user's `sub` UUID from the decoded payload, with which we will
+            # authenticate later.
+            initial_user = CustomAuthentication.authenticate(
+                self, username=email
+            )
+
+            if initial_user.sub is None:
+                # Save the `sub` to the superuser.
+                initial_user.sub = sub
+                initial_user.save()
+
+                # Login with the new superuser.
+                self.login_user(request, initial_user, "User Found")
+                return initial_user
+
+        # Authenticate with `sub` and not username, as user's can change their
+        # corresponding emails externally.
         user = CustomAuthentication.authenticate(
-            self, username=decoded_payload["email"]
+            self, user_id=sub
         )
 
         if user and user.is_active:
+            # User's are able to update their emails on login.gov
+            # Update the User with the latest email from the decoded_payload.
+            if user.username != email:
```

Comment:
If the user update's their email externally, we manually update it here so it can be used in future login attempts.